pennylane/interfaces/execution.py

```diff
@@ -310,6 +310,7 @@ def cost_fn(params, x):
     if isinstance(cache, bool) and cache:
         # cache=True: create a LRUCache object
         cache = LRUCache(maxsize=cachesize, getsizeof=lambda x: qml.math.shape(x)[0])
+        cache._persistent_cache = False

```

Comment:
Very nice solution! I only have one hesitation: if a user passes a cache object which has defined `__slots__`, or which is a C++-based class internally, then this will fail with

```
AttributeError: 'some cache' object has no attribute '_persistent_cache'
```

Reply:
Hmm yes good point! It's also suboptimal in the unlikely case of an attribute name clash 👀